### PR TITLE
Set up waitlist

### LINKED ISSUE #3: Set up waitlist
This consists in a model, named `Waiter` for instance.
It only contains a field named `email`.

Implement a form, allowing to sign up to this waitlist.

PS: There already exists a edge file corresponding to the page to implement.

### PR DESCRIPTION
**Pull request by lucaschabiron**
Close #3

### CODE REVIEW

File: web/app/Controllers/Http/WaitlistController.ts [hunk 1]
```diff
@@ -1,7 +1,29 @@
...
+import Waiter from 'App/Models/Waiter'
 
 export default class WaitlistController {
   public async show({ view }: HttpContextContract) {
     return view.render('pages/waitlist')
   }
+
+  public async store({ request, response, session }: HttpContextContract) {
+    const email = request.input('email')
+    if (!email) {
```
Comment (lucas-jacques) [inline web/app/Controllers/Http/WaitlistController.ts:11]: Use a validator instead

File: web/app/Models/Waiter.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { DateTime } from 'luxon'
+import { BaseModel, column } from '@ioc:Adonis/Lucid/Orm'
+
+export default class Waiter extends BaseModel {
+  @column({ isPrimary: true })
+  public id: number
```
Comment (lucas-jacques) [inline web/app/Models/Waiter.ts:6]: No need for an id when you extend from the ValyentBaseModel

Comment (lucas-jacques) [reply web/app/Models/Waiter.ts:6]: A cuid is generated

File: web/database/migrations/1680560694313_waiters.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import BaseSchema from '@ioc:Adonis/Lucid/Schema'
+
+export default class extends BaseSchema {
+  protected tableName = 'waiters'
+
+  public async up() {
+    this.schema.createTable(this.tableName, (table) => {
+      table.increments('id')
```
Comment (lucas-jacques) [inline web/database/migrations/1680560694313_waiters.ts:8]: When you use the ValyentBaseModel, it becomes a string
`      table.string().primary()`

File: web/resources/views/components/modals/error_modal.edge [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<div class="rounded-md bg-red-50 p-4">
+  <div class="flex">
+    <div class="flex-shrink-0">
+      <svg class="h-5 w-5 text-red-400" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
+        <path fill-rule="evenodd" d="M8.485 3.495c.673-1.167 2.357-1.167 3.03 0l6.28 10.875c.673 1.167-.17 2.625-1.516 2.625H3.72c-1.347 0-2.189-1.458-1.515-2.625L8.485 3.495zM10 6a.75.75 0 01.75.75v3.5a.75.75 0 01-1.5 0v-3.5A.75.75 0 0110 6zm0 9a1 1 0 100-2 1 1 0 000 2z" clip-rule="evenodd" />
+      </svg>
+    </div>
+    <div class="ml-3">
+      <p class="text-sm font-medium text-red-800"> {{content}}</p>
```
Comment (lucas-jacques) [inline web/resources/views/components/modals/error_modal.edge:9]: Formatting

File: web/resources/views/components/modals/success_modal.edge [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<div class="rounded-md bg-green-50 p-4">
+  <div class="flex">
+    <div class="flex-shrink-0">
+      <svg class="h-5 w-5 text-green-400" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
+        <path fill-rule="evenodd" d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.483 4.79-1.88-1.88a.75.75 0 10-1.06 1.061l2.5 2.5a.75.75 0 001.137-.089l4-5.5z" clip-rule="evenodd" />
+      </svg>
+    </div>
+    <div class="ml-3">
+      <p class="text-sm font-medium text-green-800"> {{content}}</p>
```
Comment (lucas-jacques) [inline web/resources/views/components/modals/success_modal.edge:9]: No need for two modals. Make a component for both modal to reuse html

Comment (lucas-jacques) [reply web/resources/views/components/modals/success_modal.edge:9]: Add an component class in index.css if needed

File: web/resources/views/pages/waitlist.edge [hunk 6]
```diff
@@ -16,14 +16,27 @@
           <p class="mt-3 max-w-2xl mx-auto text-xl text-gray-200 sm:mt-4">Rejoignez la liste d'attente pour être informé
             lorsque nous offrons une place dans notre alpha privée.</p>
           <form
-            class="relative bg-gray-700/10 border border-blue-200/10 w-full rounded-2xl xl:text-lg mt-8 flex items-center">
+            class="relative bg-gray-700/10 border border-blue-200/10 w-full rounded-2xl xl:text-lg mt-8 flex items-center" method="post" action="/waitlist">
+            {{ csrfField() }}
             <input type="email" name="email" id="email" placeholder="Entrez votre adresse électronique ici"
               class="w-full h-12 px-4 text-gray-200 bg-transparent border-none rounded-l-2xl focus:outline-none focus:ring-offset-gray-700 focus:ring-2 focus:ring-blue-500"><button
```
Comment (lucas-jacques) [inline web/resources/views/pages/waitlist.edge:22]: Use input component already  provided

**Review by lucas-jacques — CHANGES_REQUESTED** [review]

File: web/resources/views/components/modals/success_modal.edge (hunk 5 above)
Comment (alexisbouchez) [reply web/resources/views/components/modals/success_modal.edge:9]: These components should not be called modals, but notifications.

**Review by alexisbouchez — APPROVED** [review]

**Review by alexisbouchez — APPROVED** [review]

**Review by alexisbouchez — APPROVED** [review]

**Review by alexisbouchez — APPROVED** [review]

**Review by lucas-jacques — APPROVED** [review]

**Pull request merged**